rye/tests/common/mod.rs

```diff
@@ -56,6 +56,33 @@ toolchain = "cpython@3.12.1"
         .unwrap();
     }
 
+    // write the credentials file
+    let credentials_file = home.join("credentials");
+    if !credentials_file.is_file() {
```

Comment:
When we write the home contents how are we managing changes to those contents? I had to remove the file for this to run.

Reply:
Today the home folder is shared so there is really no way to fuck around in it between tests. One option would be to restructure the code in `common/mod.rs` to make two homes. One like used by all (concurrent), another one that uses the file lock to ensure only ever one test can use it at the time.